metric-anomaly-task-manager/src/integrationTest/java/org/hypertrace/alert/engine/metric/anomaly/task/manager/TaskManagerTest.java

```diff
@@ -45,12 +89,130 @@ public static void setup() throws Exception {
             .withStartupAttempts(2)
             .waitingFor(Wait.forListeningPort());
     kafkaZk.start();
+
+    mongo =
+        new GenericContainer<>(DockerImageName.parse("mongo:4.2.0"))
+            .withNetwork(network)
+            .withNetworkAliases("mongo")
+            .withExposedPorts(27017)
+            .withStartupAttempts(CONTAINER_STARTUP_ATTEMPTS)
+            .waitingFor(Wait.forListeningPort())
+            .withLogConsumer(logConsumer);
+    mongo.start();
+
+    configService =
+        new GenericContainer<>(DockerImageName.parse("hypertrace/config-service:main"))
+            .withNetwork(network)
+            .withNetworkAliases("config-service")
+            .withEnv("MONGO_HOST", "mongo")
+            .withExposedPorts(50101)
+            .dependsOn(mongo)
+            .withStartupAttempts(CONTAINER_STARTUP_ATTEMPTS)
+            .waitingFor(Wait.forLogMessage(".*Started admin service on port: 50102.*", 1));
+    configService.start();
+    configService.followOutput(logConsumer);
+
+    managedChannel =
+        ManagedChannelBuilder.forAddress("localhost", configService.getMappedPort(50101))
+            .usePlaintext()
+            .build();
   }
 
   @AfterAll
   public static void shutDown() {
     kafkaZk.stop();
+    mongo.stop();
+    configService.stop();
     network.close();
+    managedChannel.shutdown();
+  }
+
+  @Test
+  public void testReadRulesFromDb() throws IOException {
+    // first create rules
+    NotificationRuleConfigServiceGrpc.NotificationRuleConfigServiceBlockingStub
```

Comment:
nit: Can you move the creation of the rule to a private function? That will be easy for reading and if requires will be useful in setup or other functions.